src/components/core/geoChart/geoChartDataLayers.ts

```diff
@@ -0,0 +1,92 @@
+// (C) 2019 GoodData Corporation
+import { Execution } from "@gooddata/typings";
+import mapboxgl from "mapbox-gl";
+import {
+    DEFAULT_PUSHPIN_COLOR_PALETTE,
+    DEFAULT_PUSHPIN_COLOR_VALUE,
+    DEFAULT_PUSHPIN_OPTIONS,
+    DEFAULT_PUSHPIN_SIZE_SCALE,
+    DEFAULT_PUSHPIN_SIZE_VALUE,
+    DEFAULT_PUSHPIN_VALUE_KEY,
+} from "../../../constants/geoChart";
+import { stringToFloat } from "../../../helpers/utils";
+import { IGeoDataIndex } from "../../../interfaces/Config";
+
+function createPushPinColorOptions(
+    executionResult: Execution.IExecutionResult,
+    geoDataIndex: IGeoDataIndex,
+): mapboxgl.Expression | string {
+    const data = executionResult.data as Execution.DataValue[][];
+    const colorData = (data[geoDataIndex.color] || []).map(stringToFloat);
+
+    if (colorData.length === 0) {
+        return DEFAULT_PUSHPIN_COLOR_VALUE;
+    }
+
+    const colorMax: number = Math.max(...colorData);
+    const colorMin: number = Math.min(...colorData);
+
+    const colorsCount = DEFAULT_PUSHPIN_COLOR_PALETTE.length;
+    const colorStep = colorsCount > 0 ? (colorMax - colorMin) / colorsCount : 0;
+
+    const colorOptions: mapboxgl.Expression = ["step", ["get", DEFAULT_PUSHPIN_VALUE_KEY], "#000000"];
+    for (let index = 0; index < colorsCount; index++) {
+        let stepValue = colorMin + colorStep * index;
+        if (stepValue > colorMax) {
+            stepValue = colorMax;
+        }
+        colorOptions.push(stepValue);
+        colorOptions.push(DEFAULT_PUSHPIN_COLOR_PALETTE[index]);
```

Comment:
It could be `colorOptions.push(stepValue, DEFAULT_PUSHPIN_COLOR_PALETTE[index]);`